### PR TITLE
Added ability to save tensors

### LINKED ISSUE #2: Need API for saving to file.
Currently we only have a load function. But after training the network it would be great if we can save the generated tensor to a file.

@soumith I think we can reuse a lot of code from this repo https://github.com/MattVitelli/GRUV

### PR DESCRIPTION
**Pull request by SeanNaren**
Ported the code from [here](https://github.com/soumith/lua---audio) to save tensors! Should also close #2, let me know if any feedback

### CODE REVIEW

File: torchaudio/__init__.py [hunk 1]
```diff
@@ -30,3 +32,27 @@ def load(filename, out=None):
...
+        func = th_sox.libthsox_Byte_write_audio_file
+    elif isinstance(src, torch.CharTensor):
+        func = th_sox.libthsox_Char_write_audio_file
+    elif isinstance(src, torch.ShortTensor):
+        func = th_sox.libthsox_Short_write_audio_file
+    elif isinstance(src, torch.IntTensor):
+        func = th_sox.libthsox_Int_write_audio_file
+    elif isinstance(src, torch.LongTensor):
+        func = th_sox.libthsox_Long_write_audio_file
+    func(bytes(filepath), src, extension.replace('.', ''), sample_rate)
```
Comment (apaszke) [inline torchaudio/__init__.py]: why not just slice the first character?

File: torchaudio/__init__.py [hunk 2]
```diff
@@ -30,3 +32,27 @@ def load(filename, out=None):
...
+    elif isinstance(src, torch.ByteTensor):
+        func = th_sox.libthsox_Byte_write_audio_file
+    elif isinstance(src, torch.CharTensor):
+        func = th_sox.libthsox_Char_write_audio_file
+    elif isinstance(src, torch.ShortTensor):
+        func = th_sox.libthsox_Short_write_audio_file
+    elif isinstance(src, torch.IntTensor):
+        func = th_sox.libthsox_Int_write_audio_file
+    elif isinstance(src, torch.LongTensor):
+        func = th_sox.libthsox_Long_write_audio_file
```
Comment (apaszke) [inline torchaudio/__init__.py]: How about
```python
if not torch.is_tensor(src):
    raise ...
if not src.__module__ == 'torch':
    raise ...
typename = type(src).__name__.replace('Tensor', '')
func = getattr(th_sox, 'libthsox_{}_write_audio_file'.format(typename))
```

File: torchaudio/__init__.py [hunk 3]
```diff
@@ -30,3 +32,27 @@ def load(filename, out=None):
     func(bytes(filename), out, sample_rate_p)
     sample_rate = sample_rate_p[0]
     return out, sample_rate
+
+
+def save(filepath, src, sample_rate):
+    assert torch.is_tensor(src)
+    assert not src.is_cuda
```
Comment (apaszke) [inline torchaudio/__init__.py]: It's better to use ifs and raise proper errors. You can use assertions to check invariants, not arguments

Comment (SeanNaren) [inline torchaudio/__init__.py]: Would it make sense for me to change this in the load function as well?

Comment (apaszke) [inline torchaudio/__init__.py]: Yeah, sure

Comment (SeanNaren) [inline torchaudio/__init__.py]: Did it anyway :)

**Comment by SeanNaren** [comment]
Thanks for the feedback @apaszke, made a few changes if you could review again!

File: README.md [hunk 4]
```diff
@@ -49,3 +50,13 @@ audio.load(
 )
 ```
 
+torchaudio.save
+```
+saves a tensor into an audio file. The extension of the given path is used as the saving format.
+audio.save(
+    string, # path to file
+	tensor, # NSamples x NChannels 2D tensor
+	number, # sample_rate of the audio to be saved as
```
Comment (apaszke) [inline README.md]: it seems that you mixed up tabs and spaces here

**Pull request merged**